pkg/graveler/committed/commit_test.go

```diff
@@ -0,0 +1,339 @@
+)
+
+func TestCommitAdd(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	base := testutil.NewFakeIterator()
+	base.
+		AddRange(&committed.Range{ID: "one", MinKey: committed.Key("a"), MaxKey: committed.Key("cz"), Count: 2}).
+		AddValueRecords(makeV("a", "base:a"), makeV("c", "base:c")).
+		AddRange(&committed.Range{ID: "two", MinKey: committed.Key("d"), MaxKey: committed.Key("d"), Count: 1}).
+		AddValueRecords(makeV("d", "base:d"))
+	changes := testutil.NewValueIteratorFake([]graveler.ValueRecord{*makeV("b", "dest:b"),
+		*makeV("e", "dest:e"),
+		*makeV("f", "dest:f")})
+
+	writer := mock.NewMockMetaRangeWriter(ctrl)
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("a", "base:a")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("b", "dest:b")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("c", "base:c")))
+	writer.EXPECT().WriteRange(gomock.Eq(committed.Range{ID: "two", MinKey: committed.Key("d"), MaxKey: committed.Key("d"), Count: 1}))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("e", "dest:e")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("f", "dest:f")))
+
+	summary, err := committed.Commit(context.Background(), writer, base, changes, &committed.CommitOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, graveler.DiffSummary{
+		Count: map[graveler.DiffType]int{
+			graveler.DiffTypeAdded: 3,
+		},
+	}, summary)
+}
+
+func TestCommitChangeRangesWithinBaseRange(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	base := testutil.NewFakeIterator()
+	base.
+		AddRange(&committed.Range{ID: "outer-range", MinKey: committed.Key("k1"), MaxKey: committed.Key("k6"), Count: 2}).
+		AddValueRecords(makeV("k1", "base:k1"), makeV("k6", "base:k6")).
+		AddRange(&committed.Range{ID: "last", MinKey: committed.Key("k10"), MaxKey: committed.Key("k13"), Count: 2}).
+		AddValueRecords(makeV("k10", "base:k10"), makeV("k13", "base:k13"))
+	changes := testutil.NewValueIteratorFake([]graveler.ValueRecord{*makeV("k2", "dest:k2"),
+		*makeV("k3", "dest:k3"),
+		*makeV("k4", "dest:k4"),
+		*makeV("k5", "dest:k5")})
+
+	writer := mock.NewMockMetaRangeWriter(ctrl)
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k1", "base:k1")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k2", "dest:k2")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k3", "dest:k3")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k4", "dest:k4")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k5", "dest:k5")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k6", "base:k6")))
+	writer.EXPECT().WriteRange(gomock.Eq(committed.Range{ID: "last", MinKey: committed.Key("k10"), MaxKey: committed.Key("k13"), Count: 2}))
+	summary, err := committed.Commit(context.Background(), writer, base, changes, &committed.CommitOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, graveler.DiffSummary{
+		Count: map[graveler.DiffType]int{
+			graveler.DiffTypeAdded: 4,
+		},
+	}, summary)
+}
+
+func TestCommitBaseRangesWithinChangeRange(t *testing.T) {
```

Comment:
No such thing as "change range" in the context of commit, since the changes don't come in ranges.

Reply:
Done